torchmdnet/datasets/comp6.py

```diff
@@ -41,31 +60,49 @@ def raw_url_name(self):
 
     @property
     def raw_url(self):
-        url_prefix = 'https://raw.githubusercontent.com/isayev/COMP6/master/COMP6v1'
-        return [f'{url_prefix}/{self.raw_url_name}/{name}' for name in self.raw_file_names]
+        url_prefix = "https://raw.githubusercontent.com/isayev/COMP6/master/COMP6v1"
+        return [
+            f"{url_prefix}/{self.raw_url_name}/{name}" for name in self.raw_file_names
+        ]
+
+    @staticmethod
+    def compute_reference_energy(atomic_numbers):
+        atomic_numbers = np.array(atomic_numbers)
+        energy = sum(COMP6Base.ELEMENT_ENERGIES[z] for z in atomic_numbers)
+        return energy * COMP6Base.HARTREE_TO_EV
 
     def download(self):
         for url in self.raw_url:
             download_url(url, self.raw_dir)
 
     @property
     def processed_file_names(self):
-        return [f'{self.name}.idx.mmap',
-                f'{self.name}.z.mmap',
-                f'{self.name}.pos.mmap',
-                f'{self.name}.y.mmap',
-                f'{self.name}.dy.mmap']
+        return [
+            f"{self.name}.idx.mmap",
+            f"{self.name}.z.mmap",
+            f"{self.name}.pos.mmap",
+            f"{self.name}.y.mmap",
+            f"{self.name}.dy.mmap",
+        ]
 
     def sample_iter(self):
 
-        for path in tqdm(self.raw_paths, desc='Files'):
+        for path in tqdm(self.raw_paths, desc="Files"):
             molecules = list(h5py.File(path).values())[0].values()
 
-            for mol in tqdm(molecules, desc='Molecules', leave=False):
-                z = pt.tensor([self.ATOMIC_NUMBERS[atom] for atom in mol['species']], dtype=pt.long)
-                all_pos = pt.tensor(mol['coordinates'][:], dtype=pt.float32)
-                all_y = pt.tensor(mol['energies'][:] * self.HARTREE_TO_EV, dtype=pt.float64)
-                all_dy = pt.tensor(mol['forces'][:] * self.HARTREE_TO_EV, dtype=pt.float32)
+            for mol in tqdm(molecules, desc="Molecules", leave=False):
+                z = pt.tensor(
+                    [self.ATOMIC_NUMBERS[atom] for atom in mol["species"]],
+                    dtype=pt.long,
+                )
+                all_pos = pt.tensor(mol["coordinates"][:], dtype=pt.float32)
+                all_y = pt.tensor(
+                    mol["energies"][:] * self.HARTREE_TO_EV, dtype=pt.float64
+                )
+                all_dy = pt.tensor(
+                    mol["forces"][:] * self.HARTREE_TO_EV, dtype=pt.float32
+                )
+                all_y -= self.compute_reference_energy(z)
```

Comment:
Probably, we need to change other loader. Otherwise, it just shifts the inconsistency.

Reply:
Which? QM9q does this now and SPICE returns formation energies. We should implement it for ANI-1x as well I guess